Firmware/menu.cpp

```diff
@@ -77,6 +77,11 @@ void menu_start(void)
 
 void menu_end(void)
 {
+	if (menu_row >= LCD_HEIGHT)
```

Comment:
Should this be `if(menu_clicked)` ?

Reply:
It can't be menu_clicked because it is cleared in menu_item_ret(). I had to add this workaround since the menu_end() uses lcd_encoder, but lcd_encoder is invalid in the for loop cycle after a click event, so it would result in broken behaviour if I didn't do that extra check. I hope this made sense. The entire menu code is really confusing and I'd love to improve this MENU_BEGIN/END at some point since I see a lot of unnecessary complexity that ends up in needing these workarounds.